File: 215-kth-largest-element-in-an-array/kth-largest-element-in-an-array.py
import heapq
class Solution:
    def findKthLargest(self, nums: List[int], k: int) -> int:
        pq=[]
        for i in nums:
            heapq.heappush(pq,i)
        for i in range(len(nums)-k):
            heapq.heappop(pq)
        return heapq.heappop(pq)

File: 215-kth-largest-element-in-an-array/kth-largest-element-in-an-array.java
class Solution {
    public int findKthLargest(int[] nums, int k) {
        if(k>nums.length)return -1;
        PriorityQueue<Integer> pq=new PriorityQueue<>(Collections.reverseOrder());
        for(int v:nums)pq.offer(v);
        for(int i=0;i<k-1;i++)pq.poll();
        return pq.poll();
    }
}